data/lua/core/map.lua

```diff
@@ -167,48 +209,75 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 		end
 	end
 
-	function hex_mt:fogged_for(side)
+	---Test if the hex is under fog for a specific side
+	---@param side integer|side
+	---@return boolean
+	function hex_methods:fogged_for(side)
 		return wesnoth.sides.is_fogged(side, self)
 	end
 
-	function hex_mt:shrouded_for(side)
+	---Test if the hex is under shroud for a specific side
+	---@param side integer|side
+	---@return boolean
+	function hex_methods:shrouded_for(side)
 		return wesnoth.sides.is_shrouded(side, self)
 	end
 
-	function hex_mt:set_shrouded(side, val)
+	---Set whether the hex is shrouded for a specific side
+	---@param side integer|side
+	---@param val boolean
+	function hex_methods:set_shrouded(side, val)
 		if val then
 			wesnoth.sides.place_shroud(side, {self})
 		else
 			wesnoth.sides.remove_shroud(side, {self})
 		end
 	end
 
-	function hex_mt:set_fogged(side, val)
+	---Set whether the hex is fogged for a specific side
+	---@param side integer|side
+	---@param val boolean
+	function hex_methods:set_fogged(side, val)
 		if val then
 			wesnoth.sides.place_fog(side, {self})
 		else
 			wesnoth.sides.remove_fog(side, {self})
 		end
 	end
 
-	function hex_mt:label_for(who)
+	---Get a label placed by a specific side
```

Comment:
The code in game_lua_kernel::intf_get_label looks a lot more complex than this, it seems to be "visible to" along with handling of allied sides and recognising strings as names too.